runtime/lua/vim/diagnostic.lua

```diff
@@ -865,7 +865,16 @@ local function set_list(loclist, opts)
     end
   end
   if open then
-    api.nvim_command(loclist and 'lwindow' or 'botright cwindow')
+    if loclist then
+      api.nvim_command('lwindow')
+    else
+      -- First navigate to the diagnostics quickfix list.
+      local nr = vim.fn.getqflist({ id = _qf_id, nr = 0 }).nr
+      api.nvim_command(nr .. 'chi')
```

Comment:
best practice to use full names in code (helps with grep, etc)

```suggestion
      api.nvim_command(nr .. 'chistory')
```

Reply:
Fixed! I actually did look for the full name, but am not very good at reading vim docs. `:help chi` lead me to:

```
:[count]chi[story]      Show the list of error lists.  The current list is
```

Which I glossed over as "optional count prefix" and "optional story suffix". Oops!